python/cuml/metrics/trustworthiness.pyx

```diff
@@ -0,0 +1,110 @@
+#
+# Copyright (c) 2018-2019, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
+
+import cudf
+import numpy as np
+
+from numba import cuda
+
+from libc.stdint cimport uintptr_t
+from cuml.common.handle cimport cumlHandle
+
+cdef extern from "metrics/trustworthiness.h" namespace "ML::Metrics":
+    cdef double trustworthiness_score[T](const cumlHandle& h, T* X,
+              T* X_embedded, int n, int m, int d, int n_neighbors)
+
+def trustworthiness(X, X_embedded, handle=None, n_neighbors=5):
```

Comment:
We should provide the `should_downcast` option  here and set it to `True` by default. Take a look at the cython for  `NearestNeighbors` or `UMAP` to see how this was done.